### ISSUE TITLE
New ideas and updates

### ISSUE THREAD

**Issue by w33zy** [dasmaeh/recipepress-reloaded#39]
Hello Jan,
I have now have the time to implement some ideas and bug fixes for RPR. As such I am reaching out to you to get an idea of what new features you may have on the roadmap.

**Comment by jckoester** [comment]
Hi,

I'm sorry, but at the moment I'm awfully busy as the new school year has just started.
In summer I've played around with the idea of making the code modular. But I got stuck while thinking about the rendering/output. So no real progress there and no roadmap either.

**Comment by w33zy** [comment]
Yeah, after I sent that I remember you mentioning that you are a teacher. I ought to have reached out to you earlier during the Summer break.

For the time being I'll just submit PRs for bug fixes as I come across them.

**Comment by jckoester** [comment]
Don't worry. Give me 2 or 3 weeks and I will have some time to spend on the plugin again.

If you like, have a look at the `module_system`branch. Unfortunately it's not documented well. Once I know how to do the output rendering, I'd like to move stuff into modules.

**Comment by w33zy** [comment]
Hello Jan,
How are things with you and any coming updates for RPR? 
I have some updates and tweaks for the plugin but I am trying to avoid going ahead of your branch because I am trying avoid back-porting your additions.

**Comment by jckoester** [comment]
Well things aren't getting much better I'm afraid.
I've got far less time time to spend on RPR as I wish and there are more and more projects popping up.
Anyways, today I think I've found a way to handle layouts and modules although this is not ideal as layouts have to be aware of available modules. Not a big problem though.
My plan is to create modules for existing features like nutritional information as soon as possible and then start tidying up the code.
This does mean however code will change a lot.

If you could tell me what you are trying to achieve I might be able to change the affected parts of code either first or pretty last so we don't get into trouble there.

**Comment by w33zy** [comment]
I figured that was the case as far as your time is concerned, that's why I didn't reachout over the holidays.

Most of my fixes as to do with cleaning up the layout of the recipe templates, such as the use of `&and;` to create spaces and extra spaces between commas. Also, there is fixing the layout of recipe edit screen, the old layout of the ingredient input looked a bit 'neater', and the recipe description editor field could do with more vertical height as most of my clients write a bit before they start the recipe content.

My biggest addition would be a recipe rating field via the comment form but that would best be implemented as a module, so I've hold off on that for now.

**Comment by jckoester** [comment]
Sounds great to me.
All changes to the layout are very welcome soon. I'll wait for your adjustments before creating 'core modules' for ingredients, descriptions, ...
Once that is done, I guess the module api is stable and you can create your rating module.
Looking forward to your contributions!

**Comment by w33zy** [comment]
Great!
I will start sending PRs tomorrow.

**Comment by w33zy** [comment]
A bit off topic... but what are your considerations on Gutenberg. Personally I think it is too much work for plugin authors like yourself to convert their use of native WP metaboxes to Gutenberg blocks and layouts. 

I think Gutenberg will be a disaster as I have enough trouble as it is guiding clients on entering data in a metabox, muchless a Gutenberg block.

**Comment by jckoester** [comment]
I haven't had a closer look at Gutenberg until now. 
Although I do like the general idea behind Gutenberg, I'm afraid it will we a huge load of work to convert a plugin like RPR into something compatible to Gutenberg. I'm curtrently unsure if a modular structure will help here or make everything even worse.

**Comment by w33zy** [comment]
Having taken a close look at Gutenberg and how RPR uses custom metaboxes, it would seem the 
 WordPress and Gutenberg team expect plugins dev to rewrite the entire plugin from scratch. I mean, it's like they're nuking the whole idea of "custom metaboxes" and replacing it with "blocks".

To me, its seems too much

**Comment by w33zy** [comment]
I just saw your todo of dropping JSON-LD in an upcoming update but i don't think it's necessary as the data for it can be built in a modular way using an array and filters.

Besides, JSON-LD has basically won out as the markup of choice for metadata, as far as Google and SEO is concerned.

Take a look at this gist https://gist.github.com/w33zy/bae18d16409f80746b5acc9701b23b28

This is how WooCommerce handle data generation of JSON-LD

**Comment by jckoester** [comment]
Thank's for the hint. I guess I was thinking to modular centric.

**Comment by w33zy** [comment]
When you're ready I'll rewrite for you the template tags to use the design. ~~I am thinking of stripping the old micro-data markup from the template output but I don't know your use case for it.~~

Just saw you already have this in mind.

With modularity in mind, this could even be refactored to a class by itself

**Comment by jckoester** [comment]
I planned to create a function in the public class to render the json-ld script independently from the layout. Modules need to provide their structured data as an array, so it can be collected and then rendered.
I don't think we'll need a separate class for this.

**Comment by w33zy** [comment]
OK then, I plan on looking through the module you just added to get a better idea of how you are restructuring things.

**Comment by w33zy** [comment]
Hello again, Jan.
Would it be OK if I fork RPR and release it on the WordPress.org repo? I have a lot of ideas for making RPR better but I don't know how best to proceed.

**Comment by jckoester** [comment]
Hi,
glad that you ask.  I'll definitely won't have the time to continue developing RPR any further. If you want to take over, please do so.
Let's check out if and how a wordpress.org repo can be transferred. That would be the easiest way for you and for all users of RPR.

**Comment by w33zy** [comment]
That would be great!
I swear I woke up thinking about what to do with RPR this morning, then I got an email about the support ticket that was just created on WP.org. Hence this message.

I have been thinking about RPR a lot as the site I use it on is getting very popular these days and I am interact with a lot of recipe bloggers daily.

**Comment by w33zy** [comment]
Hey Jan,
I found the instructions for transferring plugin ownership here... https://developer.wordpress.org/plugins/wordpress-org/plugin-developer-faq/#plugin-ownership

**Comment by jckoester** [comment]
Just wanted to add you as a contributor over at wordpress.org ...
Please mail me you wordpress.org username

**Comment by w33zy** [comment]
My username on wp.org is "w33zy". I would email you but I can't find your email address again.

https://profiles.wordpress.org/w33zy
